packages/world/src/modules/init/implementations/getSignedMessageHash.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { ResourceId } from "@latticexyz/store/src/ResourceId.sol";
+
+bytes32 constant DELEGATION_TYPEHASH = keccak256(
+  "Delegation(address delegatee,bytes32 delegationControlId,bytes initCallData,uint256 nonce)"
+);
+
+function getSignedMessageHash(
+  address delegatee,
+  ResourceId delegationControlId,
+  bytes memory initCallData,
+  uint256 nonce,
+  address worldAddress
+) view returns (bytes32) {
+  bytes32 domainSeperator = keccak256(
```

Comment:
You may cache the `domainSeparator` for some gas savings. It'd be something like this:

```
constructor() {
       ....

        cachedChainId = block.chainid;
        cachedThis = address(this);
        cachedDomainSeparator = _constructDomainSeparator();
}

function _getDomainSeparator() internal view returns (bytes32) {
    if (address(this) == cachedThis && block.chainid == cachedChainId) {
        return cachedDomainSeparator;
    } else {
        return _constructDomainSeparator();
    }
}

/// @notice Construct the domain separator
function _constructDomainSeparator() internal view returns (bytes32) {
    return keccak256(abi.encode(DOMAIN_TYPEHASH, block.chainid, address(this)));
}
```

Reply:
is keccak so expensive that its worth the sstore/sread?

Reply:
Yup I saw this pattern in [OpenZeppelin's implementation](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/cryptography/EIP712.sol). It's awkward here with MUD's proxy pattern (we can't use the constructor), but definitely want to optimize this after the first pass on the PR

Reply:
You're right. It might have worked if `DOMAIN_TYPEHASH` was made of more arguments, good call.